test/pageobjects/login.page.js

```diff
@@ -34,9 +102,119 @@ class LoginPage extends Page {
     }
+    async logoClick() {
+        await this.logo.click()
+    }
+    async scrollToProduct() {
+        await this.product.scrollIntoView();
+    }
+    async clickToProduct() {
+        await this.product.click();
+    }
+    async clickToColorDropDown() {
+        await this.colorDropdown.click();
+    }
+    async selectColor(color) {
+        await this.colorDropdown.selectByVisibleText(color);
+    }
+    async clickToSizeDropdown() {
+        await this.sizeDropdown.click();
+
+    }
+    async selectSize(size) {
+        await this.sizeDropdown.selectByVisibleText(size);
+    }
+    async clicktoAddToCartButton() {
+        await this.btnAddToCart.click();
+    }
+    async clickToViewCartBtn() {
+        await this.btnViewCart.click();
+    }
+    async scrollToTblCart() {
+        await this.tblCart.scrollIntoView();
+    }
+    async clickToProceedToCheckoutbtn() {
+        await this.btnProceedToCheckout.click();
+    }
+    async clickOnFirstnameField(){
+       await this.inputFirstName.click(); 
+       
+    }
+    async enterFirstName(firstname){
+        await this.inputFirstName.setValue(firstname);
+    }
+    async clickOnLastNameField(){
+        await this.inputLastName.click(); 
+        
+     }
+     async enterLastName(lastname){
+         await this.inputLastName.setValue(lastname);
+     }
+     async clickOnCompany(){
+        await this.inputCompany.click();
+     }
+     async enterCompany(company){
+        await this.inputCompany.setValue(company);
+    }
+    async clickOnCountry(){
+        await this.countryArrow.click();
+     }
+     async enterCountry(country){
+        await this.inputCountry.selectByVisibleText(country);
+    }
+    async clickOnAddress(){
+        await this.inputBillingAddress.click(); 
+        
+     }
+     async enterAddress(address){
+         await this.inputBillingAddress.setValue(address);
+     }
+     async clickOnTown(){
+        await this.inputTown.click();
+     }
+     async enterTown(town){
+        await this.inputTown.setValue(town);
+     }
+     async clickOnCountry(){
+        await this.countryArrow.click();
+     }
+     async enterState(state){
+        await this.selectBillingState.selectByVisibleText(state);
+     }
+     async clickOnPostCode(){
+        await this.inputPostCode.click();
+     }
+     async enterPostCode(zip){
+        await this.inputPostCode.setValue(zip);
+     }
+
+     async clickOnPhone(){
+        await this.inputPhone.click();
+     }
+     async enterPhone (phone){
+        await this.inputPhone.setValue(phone);
+     }
+     async clickOnEmail(){
+        await this.inputEmail.click();
+     }
+     async enterEmail(email){
+        await this.inputEmail.setValue(email);
+     }
+     async clickOnTerms(){
+        await this.inputTerms.click();
+     }
+     async clickOnPlaceOrderBtn(){
+        await this.btnPlaceOrder.click();
+     }
+
+
+
+
+
+
 }
 
 export default new LoginPage();
```

Comment:
Here I got some kind of error.
I tried to export login.page.js as export default LoginPage. 
I also tried to import as you said, with require, but then I have this message that this isn't the way how is done with ES modules. 

So, I reversed that as it is for opportunity do discuss it further, and to learn correctly.

Reply:
You can mix export and require. Try to keep all the current logic, except exporting new object.

So only this:
export default LoginPage;

And in Smoke test you can import?

Reply:
So, without const loginPage = new LoginPage() in smoke?
I tried with that and it was an error. 
But I'll try again.

Reply:
I tried various solutions but I still get the same errors. 
It isn't allowed to use require because of ES module standards.
With this export default new XyPage();
and import in test, I don't get any errors. 
But I'm looking forward to learn if there is another way for writing this.